File: app/src/main/java/com/lazylee/lzywanandroid/activity/web/WebActivity.java
package com.lazylee.lzywanandroid.activity.web;


import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.lazylee.lzywanandroid.R;



public class WebActivity extends AppCompatActivity {

    private static final String TAG = WebActivity.class.getSimpleName();

    private WebView mWebView;

    private boolean jsEnable = true;

    private void initViews(){
        mWebView = findViewById(R.id.webView);
        setDefaultWebSetting(mWebView);
        mWebView.setWebViewClient(new WebViewClient());
        mWebView.setWebChromeClient(new WebChromeClient());


    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.web_activity);
        Intent intent = getIntent();
        String link = intent.getStringExtra("link");

        initViews();
        mWebView.loadUrl(link);
    }

    private void setDefaultWebSetting(WebView webView){
        WebSettings webSettings = webView.getSettings();
        webSettings.setJavaScriptEnabled(jsEnable);
        //5.0以上开启混合模式加载
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP){
            webSettings.setMixedContentMode(WebSettings.MIXED_CONTENT_ALWAYS_ALLOW);
        }
        webSettings.setLoadWithOverviewMode(true);
        webSettings.setUseWideViewPort(true);
        //允许js代码
        webSettings.setJavaScriptEnabled(jsEnable);
        //允许SessionStorage/LocalStorage存储
        webSettings.setDomStorageEnabled(true);
        //禁用放缩
        webSettings.setDisplayZoomControls(false);
        webSettings.setBuiltInZoomControls(false);
        //禁用文字缩放
        webSettings.setTextZoom(100);
        //10M缓存，api 18后，系统自动管理。
        webSettings.setAppCacheMaxSize(10 * 1024 * 1024);
        //允许缓存，设置缓存位置
        webSettings.setAppCacheEnabled(true);
        //允许WebView使用File协议
        webSettings.setAllowFileAccess(true);
        //不保存密码
        webSettings.setSavePassword(false);
        //自动加载图片
        webSettings.setLoadsImagesAutomatically(true);
    }
}

File: app/src/main/java/com/lazylee/lzywanandroid/ui/activity/web/WebActivity.kt
package com.lazylee.lzywanandroid.ui.activity.web


import android.content.Intent
import android.os.Build
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.webkit.WebChromeClient
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient

import com.lazylee.lzywanandroid.R


class WebActivity : AppCompatActivity() {

    private var mWebView: WebView? = null

    private val jsEnable = true

    private fun initViews() {
        mWebView = findViewById(R.id.webView)
        setDefaultWebSetting(mWebView!!)
        mWebView!!.webViewClient = WebViewClient()
        mWebView!!.webChromeClient = WebChromeClient()


    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.web_activity)
        val intent = intent
        val link = intent.getStringExtra("link")

        initViews()
        mWebView!!.loadUrl(link)
    }

    private fun setDefaultWebSetting(webView: WebView) {
        val webSettings = webView.settings
        webSettings.javaScriptEnabled = jsEnable
        //5.0以上开启混合模式加载
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            webSettings.mixedContentMode = WebSettings.MIXED_CONTENT_ALWAYS_ALLOW
        }
        webSettings.loadWithOverviewMode = true
        webSettings.useWideViewPort = true
        //允许js代码
        webSettings.javaScriptEnabled = jsEnable
        //允许SessionStorage/LocalStorage存储
        webSettings.domStorageEnabled = true
        //禁用放缩
        webSettings.displayZoomControls = false
        webSettings.builtInZoomControls = false
        //禁用文字缩放
        webSettings.textZoom = 100
        //10M缓存，api 18后，系统自动管理。
        webSettings.setAppCacheMaxSize((10 * 1024 * 1024).toLong())
        //允许缓存，设置缓存位置
        webSettings.setAppCacheEnabled(true)
        //允许WebView使用File协议
        webSettings.allowFileAccess = true
        //不保存密码
        webSettings.savePassword = false
        //自动加载图片
        webSettings.loadsImagesAutomatically = true
    }

    companion object {

        private val TAG = WebActivity::class.java!!.getSimpleName()
    }
}
